static/src/js/components/SpatialDisplay/SpatialDisplay.js

```diff
@@ -291,6 +300,49 @@ class SpatialDisplay extends Component {
           title="Rectangle"
         />
       ))
+    } else if (((shapefile.shapefileName || shapefile.shapefileId || shapefile.shapefileError)
+      && !drawingNewLayer)
+      || drawingNewLayer === 'shapefile') {
+      const {
```

Comment:
While it's pretty 'readable' can we add some comments here, this is a big ol' conditional.